wavefront/server/modules/voice_agents_module/voice_agents_module/services/tool_service.py

```
@@ -0,0 +1,538 @@
+                        ]
+                # else: new credentials provided, use them
+
+            # Check name uniqueness if name is being updated
+            if 'name' in update_data and update_data['name'] != tool.name:
+                existing = await self.tool_repository.find_one(
+                    name=update_data['name'], is_deleted=False
+                )
+                if existing:
+                    raise ValueError(
+                        f'Tool with name "{update_data["name"]}" already exists'
+                    )
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)